### ISSUE TITLE
Training SRDCNN

### ISSUE THREAD

**Issue by Albert22dai** [cabooster/SRDTrans#11]
Thank you for your amazing work. You have mentioned SRDCNN in your article, and I tried to reproduce SRDCNN with SRD in combination with deepcadRT, but the result is bad. Here are my training parameters:
Training parameters -----> 
{'overlap_factor': 0.125, 'datasets_path': 'datasets/DataForPytorch', 'n_epochs': 20, 'fmap': 16, 'output_dir': './results', 'pth_dir': './pth', 'onnx_dir': './onnx', 'batch_size': 1, 'patch_t': 464, 'patch_x': 64, 'patch_y': 64, 'gap_y': 56, 'gap_x': 56, 'gap_t': 406, 'lr': 5e-05, 'b1': 0.5, 'b2': 0.999, 'GPU': '0', 'ngpu': 1, 'num_workers': 0, 'scale_factor': 1, 'train_datasets_size': 1200, 'select_img_num': 100000, 'test_datasize': 1000, 'increase_ratio': 2, 'visualize_images_per_epoch': True, 'save_test_images_per_epoch': True, 'colab_display': False, 'result_display': ''}
Image list for training -----> 
Total stack number ----->  1
Noise image name ----->  noisy_6000frames.tif
Noise image shape ----->  (6000, 512, 512)
The loss function i used is L2+Lreg, I have only one GPU. Could you please tell me the specific parameters and loss function for training SRDCNN? How do I solve this problem?
![SRDCNN](https://github.com/cabooster/SRDTrans/assets/138970978/29bfbfa4-caec-4744-a1e3-b6546d2fb455)

**Comment by Albert22dai** [comment]
I tried to expand the batch_size, overlap_factor and epoch, here are the parameters:
{'overlap_factor': 0.5, 'datasets_path': 'datasets/DataForPytorch', 'n_epochs': 100, 'fmap': 64, 'output_dir': './results', 'pth_dir': './pth', 'onnx_dir': './onnx', 'batch_size': 4, 'patch_t': 128, 'patch_x': 128, 'patch_y': 128, 'gap_y': 64, 
'gap_x': 64, 'gap_t': 64, 'lr': 0.0002, 'b1': 0.5, 'b2': 0.999, 'GPU': '0', 'ngpu': 1, 'num_workers': 0, 'scale_factor': 1, 'train_datasets_size': 2000, 'select_img_num': 100000, 'test_datasize': 1000, 'increase_ratio': 2, 'visualize_images_per_epoch': False, 'save_test_images_per_epoch': True, 'colab_display': False, 'result_display': ''}
loss function is L1loss + L2loss + Lreg, It seems that the tiling effect goes away as the epoch gets bigger, but there is still a lot of noise. I don't know what went wrong and would appreciate your response.

**Comment by Albert22dai** [comment]
I tried to expand the batch_size, overlap_factor and epoch, here are the parameters:
{'overlap_factor': 0.5, 'datasets_path': 'datasets/DataForPytorch', 'n_epochs': 100, 'fmap': 64, 'output_dir': './results', 'pth_dir': './pth', 'onnx_dir': './onnx', 'batch_size': 4, 'patch_t': 128, 'patch_x': 128, 'patch_y': 128, 'gap_y': 64,
'gap_x': 64, 'gap_t': 64, 'lr': 0.0002, 'b1': 0.5, 'b2': 0.999, 'GPU': '0', 'ngpu': 1, 'num_workers': 0, 'scale_factor': 1, 'train_datasets_size': 2000, 'select_img_num': 100000, 'test_datasize': 1000, 'increase_ratio': 2, 'visualize_images_per_epoch': False, 'save_test_images_per_epoch': True, 'colab_display': False, 'result_display': ''}
loss function is L1loss + L2loss + Lreg, It seems that the tiling effect goes away as the epoch gets bigger, but there is still a lot of noise. I don't know what went wrong and would appreciate your response.
![屏幕截图 2024-05-31 235945](https://github.com/cabooster/SRDTrans/assets/138970978/8192fda0-9de8-4b85-a58b-464436193622)

**Comment by cabooster** [comment]
I think there's no need to train 100 epochs. Did you try the default hyperparameters?

**Comment by Albert22dai** [comment]
Thank you for your reply. In my second attempt, most of the default hyperparameters were retained, train_datasets_size was set to 2000, lr was set to 0.0002. Maybe I need to change the loss function. I used loss2neighbor_1 + loss2neighbor_2 +0.5*(loss_reg_neighbor_1 + loss_reg_neighbor_2). Could you tell me what loss function you used to train SRDCNN?
Here is the output of the 30 epoch:
![屏幕截图 2024-06-02 121131](https://github.com/cabooster/SRDTrans/assets/138970978/f1cd3681-c6bd-44a3-9ed3-5b911b46e2d9)

**Comment by cabooster** [comment]
The loss is L1+L2, as formulated in the paper.
![image](https://github.com/cabooster/SRDTrans/assets/71710781/1c601265-b733-4ebd-9fe9-14e8c3bf182e)

**Comment by Albert22dai** [comment]
I set the hyperparameters to the default values in SRDTrans to train the SRDCNN, and the results are similar to before. As the number of epochs increases, the tiling effect disappears, but there is still a lot of noise.
![屏幕截图 2024-06-05 183253](https://github.com/cabooster/SRDTrans/assets/138970978/f319dcf0-94cd-449c-9c69-f7c7d0307dc9)

**Comment by cabooster** [comment]
Did you compare the image before and after denoising? The SNR has been improved though there's some remaining noise.

**Comment by Albert22dai** [comment]
Yes, the SNR has been improved but there's some remaining noise, which is not wanted. So I tried to change the loss function, lr, and hopefully make an improvement. Also, I tried to use SRDTrans to denoise the same image, using the default parameter. The outcome still has noise.
![Uploading 屏幕截图 2024-06-11 225906.png…]()

**Comment by Albert22dai** [comment]
Yes, the SNR has been improved but there's some remaining noise, which is not wanted. So I tried to change the loss function, lr, and hopefully make an improvement. Also, I tried to use SRDTrans to denoise the same image, using the default parameter. The outcome still has noise.
![屏幕截图 2024-06-11 225906](https://github.com/cabooster/SRDTrans/assets/138970978/71def662-ff0e-4a91-8ea3-553f352a438d)

**Comment by careymedsci** [comment]
> Thank you for your reply. In my second attempt, most of the default hyperparameters were retained, train_datasets_size was set to 2000, lr was set to 0.0002. Maybe I need to change the loss function. I used loss2neighbor_1 + loss2neighbor_2 +0.5*(loss_reg_neighbor_1 + loss_reg_neighbor_2). Could you tell me what loss function you used to train SRDCNN? Here is the output of the 30 epoch: ![屏幕截图 2024-06-02 121131](https://github.com/cabooster/SRDTrans/assets/138970978/f1cd3681-c6bd-44a3-9ed3-5b911b46e2d9)

Hi Albert, I saw you used the loss function with L_1+L_2 + 0.5*(L_1+L_2), could you please tell me what value you use finally? I used the 0.5L_1+0.5L_2, the results are not good, so I am considering use the value you used here.
